src/inmanta/logging.py

```diff
@@ -0,0 +1,249 @@
+            cls.set_logfile_location(options.log_file)
+            formatter = logging.Formatter(fmt="%(asctime)s %(levelname)-8s %(name)-10s %(message)s")
+            cls.set_log_formatter(formatter)
+            cls.set_log_level(options.log_file_level, cli=False)
+        else:
+            if options.timed:
+                formatter = InmantaLogs._get_log_formatter_for_stream_handler(timed=True)
+                cls.set_log_formatter(formatter)
+            cls.set_log_level(str(options.verbose))
+
+    @classmethod
+    def set_log_level(cls, inmanta_log_level: str, cli: bool = True) -> None:
+        """
+        Set the logging level. A handler should have been created before
+        below the supported inmanta log levels and there equivalent in python logging:
+            "0": logging.ERROR,
+            "1": logging.WARNING,
+            "2": logging.INFO,
+            "3": logging.DEBUG,
+            "4": 2,
+            "ERROR": logging.ERROR,
+            "WARNING": logging.WARNING,
+            "INFO": logging.INFO,
+            "DEBUG": logging.DEBUG,
+            "TRACE": 2,
+
+        :param inmanta_log_level: The inmanta logging level
+        :param cli: True if the logs will be outputed to the CLI.
+        """
+        if not cls._handler:
+            raise Exception(
+                "No handler to apply options to. Please use the create_default_handler method before calling this one"
+            )
+        # maximum of 4 v's
+        if inmanta_log_level.isdigit() and int(inmanta_log_level) > 4:
+            inmanta_log_level = "4"
+
+        # The minimal log level on the CLI is always WARNING
+        if cli and inmanta_log_level == "ERROR" or (inmanta_log_level.isdigit() and int(inmanta_log_level) < 1):
+            inmanta_log_level = "WARNING"
+
+        # Converts the Inmanta log level to the Python log level
+        python_log_level = log_levels[inmanta_log_level]
+        cls._handler.setLevel(python_log_level)
+
+    @classmethod
+    def set_log_formatter(cls, formatter: logging.Formatter) -> None:
+        """
+        Set the log formatter. A handler should have been created before
+
+        :param formatter: The log formatter.
+        """
+        if not cls._handler:
+            raise Exception(
+                "No handler to apply options to. Please use the create_default_handler method before calling this one"
+            )
+        cls._handler.setFormatter(formatter)
+
+    @classmethod
+    def set_logfile_location(cls, location: str) -> None:
+        """
+        Set the location of the log file. Be careful that this function will replace the current handler with a new one
+        This means that configurations done on the previous handler will be lost.
+
+        :param location: The location of the log file.
+        """
+        file_handler = logging.handlers.WatchedFileHandler(filename=location, mode="a+")
+        if cls._handler:
+            logging.root.removeHandler(cls._handler)
```

Comment:
Shouldn't we call `close()` on the old handler? Otherwise we don't close the file if the old handler would be a FileHandler for example.